services/customdestinationmanager/customdestinationmanager.go

```diff
@@ -120,17 +118,20 @@ func (customManager *CustomManagerT) SendData(jsonData json.RawMessage, sourceID
 					logger.Infof("[%s Destination manager] created new producer: %v for destination: %s and source %s", customManager.destination, producer, destID, sourceID)
 				}
 			}
-			producerCreationLock.Unlock()
+			producerLock.Unlock()
 		}
 	} else {
-		producerCreationLock.RUnlock()
+		producerLock.RUnlock()
 	}
 
+	producerLock.RLock()
```

Comment:
Instead of trying to use the same lock for this package, we should have a separate lock per streaming destination.

Reply:
Right now, we can only send 1 event at a time across streaming destinations.

Reply:
done.